Library/Homebrew/cask/cmd/upgrade.rb

```diff
@@ -41,12 +42,16 @@ def run
           .map { |(old_cask, new_cask)| "#{new_cask.full_name} #{old_cask.version} -> #{new_cask.version}" }
           .join(", ")
 
-        upgradable_casks.each do |(old_cask, new_cask)|
-          begin
-            upgrade_cask(old_cask, new_cask)
-          rescue CaskError => e
-            caught_exceptions << e
-            next
+        if dry_run?
+          puts "Dry run: did not upgrade anything."
```

Comment:
`return puts "Dry run: did not upgrade anything." if dry_run?` and can skip the `else`